### PR TITLE
LP-595 Further changes for reloading general partner person data

### PR DESCRIPTION
**Pull request by mwestacott**
### JIRA link

https://companieshouse.atlassian.net/browse/LP-595

### Change description

* Remove former names when appropriate, if patching data
* Reload data during update upon errors
* Deal with edge cases when populating field data
* Move custom back link behaviour to the correct page
* Unit tests added and improved

### Work checklist

- [X] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/views/general-partner-usual-residential-address-territory-choice.njk [hunk 1]
```diff
@@ -1,5 +1,23 @@
 {% extends "layout.njk" %}
 
+{% block backLink %}
+  {% if props.data.generalPartner.data.forename %}
```
Comment (markpit) [inline src/views/general-partner-usual-residential-address-territory-choice.njk:4]: i think we should be passing the cache into the template so we can check if they selected uk or overseas instead of forename

Comment (mwestacott) [reply src/views/general-partner-usual-residential-address-territory-choice.njk:4]: This is @lduranteau 's code that I simply moved, but I think it's correct - it's deciding whether to go back to the General Partner Person or Legal Entity page, hence the check against forename.

(It's the Back link - so user may not have even selected UK or Overseas yet...)

**Review by markpit — APPROVED** [review]
